src/libsam3/libsam3.c

```diff
@@ -716,20 +716,23 @@ int sam3GenerateKeys(Sam3Session *ses, const char *hostname, int port,
       return -1;
     }
     //
-    if (sam3tcpPrintf(fd, "DEST GENERATE %s\n", sigtypes[sigType]) >= 0) {
-      if ((rep = sam3ReadReply(fd)) != NULL &&
-          sam3IsGoodReply(rep, "DEST", "REPLY", NULL, NULL)) {
-        const char *pub = sam3FindField(rep, "PUB"),
-                   *priv = sam3FindField(rep, "PRIV");
-        //
-        if (pub != NULL && sam3CheckValidKeyLength(pub) && priv != NULL &&
-            strlen(priv) >= SAM3_PRIVKEY_MIN_SIZE) {
-          strcpy(ses->pubkey, pub);
-          strcpy(ses->privkey, priv);
-          res = 0;
-        }
-      }
+    if (sam3tcpPrintf(fd, "DEST GENERATE\n") < 0) {
```

Comment:
Aren't you missing the sign type here?

(Sorry for pointing this out, so late)

Reply:
Yup, looks like in this case we would be limited to the defaults. I'll work something up to fix it.